openbb_platform/providers/fmp/openbb_fmp/models/form_13FHR.py

```diff
@@ -0,0 +1,138 @@
+    )
+
+
+class FMPForm13FHRFetcher(
+    Fetcher[
+        FMPForm13FHRQueryParams,
+        List[FMPForm13FHRData],
+    ]
+):
+    """Fetches and transforms data from the Form 13f endpoints."""
+
+    @staticmethod
+    def transform_query(params: Dict[str, Any]) -> FMPForm13FHRQueryParams:
+        """Transform the query params."""
+        return FMPForm13FHRQueryParams(**params)
+
+    @staticmethod
+    async def aextract_data(
+        query: FMPForm13FHRQueryParams,
+        credentials: Optional[Dict[str, str]] = None,
+        **kwargs: Any,
+    ) -> List[Dict]:
+        """Return the raw data from the Form 13f endpoint."""
+        api_key = credentials.get("fmp_api_key") if credentials else ""
+        if query.symbol.isnumeric():
+            cik = query.symbol
+        else:
+            try:
+                profile = await FMPEquityProfileFetcher.fetch_data({"symbol": query.symbol}, {"fmp_api_key": api_key})
+                cik = getattr(profile[0], "cik", "")
+                if not cik:
+                    raise OpenBBError(f"Invalid symbol -> {query.symbol}")
+            except OpenBBError as e:
+                raise OpenBBError(f"Invalid symbol -> {query.symbol} -> {e}")
+        date_url = create_url(
+            3,
+            f"form-thirteen-date/{cik}",
+            api_key,
+            query,
+            exclude=["symbol", "limit"],
+        )
+        dates = await amake_request(date_url, response_callback=response_callback, **kwargs)
+        if not dates:
+            raise EmptyDataError("No data returned for the given symbol.")
+        if not dates or len(dates) == 0:
+            warn(f"Symbol Error: No data found for symbol {cik}")
+        results: List[Dict] = []
+
+        async def get_one(date):
+            """Get data for the given date."""
+            date = datetime.strptime(date, "%Y-%m-%d").date()
+            query.date = date
+            url = create_url(
+                3,
+                f"form-thirteen/{cik}",
+                api_key,
+                query,
+                exclude=["symbol", "limit"],
+            )
+            result = await amake_request(url, response_callback=response_callback, **kwargs)
+            if not result or len(result) == 0:
+                warn(f"Symbol Error: No data found for symbol {symbol}")
+            if result:
+                results.extend(result)
+
+        await asyncio.gather(*[get_one(date) for date in dates])
+
+        if not results:
+            raise EmptyDataError("No data returned for the given symbol.")
+        if query.limit:
+            return results[:query.limit]
+        return results
+
+    @staticmethod
+    def transform_data(
+        query: FMPForm13FHRQueryParams, data: List[Dict], **kwargs: Any
+    ) -> List[FMPForm13FHRData]:
+        """Return the transformed data."""
+        if query.symbol.isnumeric():
+            [d.pop("cik", None) for d in data]
+        return [FMPForm13FHRData(**d) for d in data]
```

Comment:
Here we are iterating twice, we should do everything in one pass.
```suggestion
            [d.pop("cik", None) for d in data]
        return [FMPForm13FHRData(**{k: v for k, v in d.items if k != "cik"}) for d in data]
```